### ISSUE TITLE
Add a means of configuring the Jersey Client produced by ResourceTestRule

### ISSUE THREAD

**Issue by evnm** [dropwizard/dropwizard#1416]
Unless I'm mistaken, there isn't a way to inject functionality into [the `configureClient(ClientConfig config)` method of `ResourceTestRule`](https://github.com/dropwizard/dropwizard/blob/cc5f7a8eef8621cfc8d2ff012b525069d032238a/dropwizard-testing/src/main/java/io/dropwizard/testing/junit/ResourceTestRule.java#L215-L220). In cases where tests require custom Jersey client configuration (e.g. when [testing PATCH endpoints](http://stackoverflow.com/questions/29499106/using-patch-with-jersey-client-api-for-unit-testing)), the developer isn't left with a straightforward recourse.

I'd like to propose adding a method to `ResourceTestRule.Builder` akin to the following:

```
public Builder setClientConfigurator(j.u.function.Consumer<ClientConfig> configurator) {
    this.configurator = configurator;
    return this;
}
```

This `configurator` function would then be applied in the `configureClient` method linked-to above. Thus code like the following could be used to configure clients used in tests:

```
@ClassRule
public static final ResourceTestRule resources = ResourceTestRule.builder()
    .addConfigurator(clientConfig ->
        clientConfig.property(HttpUrlConnectorProvider.SET_METHOD_WORKAROUND, true));
    ...
```

This feature would be included in the 1.0.0 milestone, given the use of the Java 8 Consumer interface.

**Comment by arteam** [comment]
Yes, I remember I wanted similar functionality other day and I didn't found a way to configure `JerseyClient`. I ended up configured `WebTarget` instead. I would welcome such contribution. 

One point, though: I've got a pull request brewing [here](https://github.com/arteam/dropwizard/commits/refactor_resource_test) about refactoring `ResourceTestRule`. Currently this class is quite a mess and I would like to clean up it beforehand. I am going to send a pull request today, so after that, I think, you may start to work on this feature.

**Comment by arteam** [comment]
See #1418

**Comment by jplock** [comment]
Closing as #1418 was merged

**Comment by arteam** [comment]
No, this issue should not be closed. #1418 is just a foundation for a pull request to implement a solution for this issue.

**Comment by jplock** [comment]
Oh whoops, sorry about that.

**Comment by jplock** [comment]
Ok, I think this can be closed now that #1421 has been merged.  :)
